src/arithmetic/arithmetic_expression.c

```diff
@@ -499,36 +492,45 @@ SIValue AR_TOLOWER(SIValue *argv, int argc) {
     assert(argc == 1);
 
     if(SIValue_IsNull(argv[0])) return SI_NullVal();
-    char *original = argv[0].stringval.str;
-    size_t lower_len = argv[0].stringval.len + 1;
-    char *lower = calloc(lower_len, sizeof(char));
+    char *original = argv[0].stringval;
+    size_t lower_len = strlen(argv[0].stringval) + 1;
+    char lower[lower_len];
```

Comment:
There's no promise that the last char in lower array is 0,
this is why I've used calloc.
Initialise lower:
lower[lower_len] = {0};

Reply:
This promise is made in `_toLower`:
```
    for(; i < str_len; i++) lower[i] = tolower(str[i]);
    lower[i] = 0;
```
Variable-length arrays cannot be initialized, we would have to use `memset`, but I don't believe that is necessary here.

I'm happy to make changes - some of the code around these functions definitely seems suboptimal - but the proposals here are neither new nor untested. All of them were reviewed in https://github.com/swilly22/redis-graph/pull/87